### ISSUE TITLE
SphinxQL Query Charset

### ISSUE THREAD

**Issue by coolswitcher** [sequelpro/sequelpro#2384]
When i'm run sphinxql query, i'm got output in wrong charset:
<img width="1030" alt="screenshot 2016-01-25 11 52 58" src="https://cloud.githubusercontent.com/assets/646663/12546470/0756f11e-c35b-11e5-8a05-3730a6ae0644.png">
Server variables of sphinx server:
<img width="314" alt="screenshot 2016-01-25 11 59 59" src="https://cloud.githubusercontent.com/assets/646663/12546496/3b83d0a6-c35b-11e5-8e1f-b1ed39236174.png">
Source mysql database in utf8 with collocation utf8_general_ci

**Comment by dmoagx** [comment]
Open Console.app and search for output from Sequel Pro.

**Comment by coolswitcher** [comment]
I found these errors:

25.01.16 12:42:20,477 Sequel Pro[11570]: SecTaskLoadEntitlements failed error=22

25.01.16 11:57:59,000 kernel[0]: Sandbox: Mail(556) deny(1) file-read-data /Applications/Sequel Pro.app/Contents/Library/QuickLook/Sequel Pro.qlgenerator/Contents

25.01.16 11:34:42,915 Sequel Pro[10995]: -[NSAlert(SPAlertDebug) sp_buttonPressed:] of <NSAlert: 0x10b5d58a0> title=
MySQL Error
text=
Unknown collation: ''utf8_general_ci''

25.01.16 11:43:14,757 Sequel Pro[10995]: -[NSAlert(SPAlertDebug) sp_buttonPressed:] of <NSAlert: 0x11790d8c0> title=
Ошибка
text=
sphinxql: syntax error, unexpected IDENT, expecting ATTRIBUTES or RAMCHUNK or RTINDEX near 'PRIVILEGES'

**Comment by dmoagx** [comment]
Do you have anything set in Database > Charset for display?
Also look into Sequel Pros Query console and check for the most recent `SET NAMES...` query that appears before loading your table.

**Comment by coolswitcher** [comment]
In Database > Charset is set UTF-8 Unicode BMP and i can't change this
In console i haven't SET NAMES queries, i'm try run:
SET NAMES utf8;
SET CHARACTER SET utf8;

No effect:(

**Comment by dmoagx** [comment]
I guess because Sphinx does not return data for the regular charset queries, Sequel Pro assumes that it is a very old MySQL version without charset support and falls back to `latin1`.

**Comment by dmoagx** [comment]
Thinking about it, I already mentioned in c7b5e880003153e599a452151ee1f3a3e40aff4c that SP will treat Sphinx as a MySQL 2.x server because of the version number.

But we have this code:
```objc
- (BOOL)setEncoding:(NSString *)theEncoding
{

	// MySQL versions prior to 4.1 don't support encoding changes; return NO on those
	// versions.
	if (![self serverVersionIsGreaterThanOrEqualTo:4 minorVersion:1 releaseVersion:0]) {
		return NO;
	}
```
+ the fallback encoding is `latin1` for backwards compatibility.

Sequel Pro also reads the `character_set_results` and `character_set` server variables to determine the encoding but it looks like Sphinx uses the other two.

**Comment by dmoagx** [comment]
I've added some code in 6457da276ba4a46dcf7e52c0b25edb42d751a13e which might fix this.
Please try a recent nightly build.

**Comment by coolswitcher** [comment]
Thanks for fix :) Now all works fine!
I'm have testing on sphinx ver 2.2.9

**Comment by dmoagx** [comment]
Issue fixed in 1.1.1
See http://www.sequelpro.com/blog/2016.02/sequel-pro-1-1-1/
